src/client.ts

```diff
@@ -159,7 +159,7 @@ export class Client {
    * @throws {@link ClientClosedError} if a query is issued after the client is closed.
    * due to an internal error.
    */
-  async query<T = any>(
+  async query<T extends QueryValue = any>(
```

Comment:
hypothetically a user could build their own types that match the output shape of the query API.

We can discuss the pros / cons of leaving this as it was to support that offline.

Reply:
```suggestion
  async query<T extends QueryValue>(
```

After working more with the new `client.paginate` I don't think default type parameter is a good idea.

But yes, let's catch up offline.